backend/data_preparation/dumper/fire_dumper.py

```diff
@@ -0,0 +1,438 @@
+"""
+@author: Scarlett Zhang
+This file has 2 classes:
+InvalidRecordError
+FireDumper
+"""
+from typing import List, Dict, Tuple
+import logging
+import rootpath
+from backend.data_preparation.dumper.dumperbase import DumperBase
+from backend.data_preparation.connection import Connection
+from typing import Set, Any
+from backend.data_preparation.crawler.fire_crawler import FireEvent
+
+rootpath.append()
+
+
+logger = logging.getLogger('TaskManager')
+
+
+class InvalidRecordError(Exception):
+    pass
+
+
+class FireDumper(DumperBase):
+    """
+    Fire Dumper class
+    In this function, there are mainly operations on 3 tables:
+    fire: schema: name, if_sequence, agency, state,id, time, geom_full, geom_1e4,geom_1e3, geom_1e2,geom_center, area.
+          primary key: (name, time)
+    (records in fire is only smaller fire event records that represent a single fire polygon record at a certain moment)
+    fire_history: schema: id, year, state, name, url.
+          primary key: id
+    (records in fire_history is a fire that takes a whole webpage on rmgsc, so 'name' here is the urlname)
+    fire_merged: schema: name, if_sequence, agency, state, id,start_time, end_time, geom_full, geom_1e4, geom_1e3,
+    geom_center, max_area
+          primary key: id
+    (records in fire_merged is a fire that takes a whole webpage on rmgsc, start_time is from the first fire record and
+    end_time is from the last fire record)
+    """
+
+    # code for checking if a database exists
+    SQL_CHECK_IF_TABLE_EXISTS = 'SELECT table_name FROM information_schema.TABLES WHERE table_name = \'{}\''
+
+    SQL_CHECK_IF_FIRE_TABLE_EXISTS = 'SELECT table_name FROM information_schema.TABLES WHERE table_name = \'fire\''
+    # SQL_CHECK_IF_FIRE_TABLE_EXISTS: check if "fire" table exists
+    SQL_CHECK_IF_HISTORY_TABLE_EXISTS = 'SELECT table_name FROM information_schema.TABLES WHERE table_name = \'fire_history\''
+    # SQL_CHECK_IF_HISTORY_TABLE_EXISTS: check if "fire_history" table exists
+    SQL_CHECK_IF_FIRE_MERGED_TABLE_EXISTS = 'SELECT table_name FROM information_schema.TABLES WHERE table_name = \'fire_merged\''
+    # SQL_CHECK_IF_FIRE_MERGED_TABLE_EXISTS: check if "fire_merged" table exists
+
+    # code for creating a table if it doesn't exist
```

Comment:
hmm, why not just fetch all table names as a tuple once (so it only sends one query to database) and then in python you can check whether a table name (str) is in the tuple of tables name. this way it saves database traffic.